tests/test_models_layers.py

```diff
@@ -187,3 +187,34 @@ def forward(self, x: LayerRef) -> LayerRef:
 
   assert "TestBlock" in net_dict and "TestBlock_0" in net_dict
   assert_equal(net_dict["TestBlock_0"]["reuse_params"], "TestBlock")
+
+
+def test_multiple_returns():
+  class _SubNet(Module):
+    def __init__(self):
+      super().__init__()
+      self.linear = Linear(n_out=42, activation=None)
```

Comment:
I would use smaller dimensions for a test (n_out < 20 or so).
Also, this is already slightly more complex than I suggested. I suggested it even without this `linear`. But anyway, this is also fine.

Reply:
Actually, I think removing this `linear` might trigger some strange other case. So leave it in (but maybe change dim).
```suggestion
      self.linear = Linear(n_out=13, activation=None)
```